### PR TITLE
tctl: Respect TELEPORT_HOME value when grabbing profile

### LINKED ISSUE #12308: `tctl` cannot find profile when TELEPORT_HOME is set and ~/.tsh is empty
Expected behavior:

#11503 was merged to make `tctl` respect the TELEPORT_HOME environment variable, but it is still not able to load the profile when TELEPORT_HOME is set and ~/.tsh is empty.

#11150 is also relevant.

https://github.com/gravitational/teleport/blob/master/tool/tctl/common/tctl.go#L329 may need to pass the variable rather than an empty string.

Current behavior:

tctl unable to load profile

Bug details:
- Teleport version - 9.1.2
- Recreation steps
- Debug logs

gz#4950

### PR DESCRIPTION
**Pull request by EdwardDowling**
Also when grabbing local key store

Closes https://github.com/gravitational/teleport/issues/12308

before:
tctl does not correctly load profile when TELEPORT_HOME is set to a non default location but ~/.tsh is empty
after:
value of TELEPORT_HOME is passed correctly to load profile

### CODE REVIEW

File: tool/tctl/common/tctl.go [hunk 1]
```diff
@@ -346,7 +345,7 @@ func loadConfigFromProfile(ccf *GlobalCLIFlags, cfg *service.Config) (*authclien
 	if err := c.LoadProfile(cfg.TeleportHome, proxyAddr); err != nil {
 		return nil, trace.Wrap(err)
 	}
-	keyStore, err := client.NewFSLocalKeyStore(c.KeysDir)
+	keyStore, err := client.NewFSLocalKeyStore(cfg.TeleportHome)
```
Comment (zmb3) [inline tool/tctl/common/tctl.go:348]: I'm not convinced that this doesn't break something.

What I would do is leave this the way it was (using `c.KeysDir`) and update `c.LoadProfile` (if necessary) to make sure it sets `c.KeysDir` appropriately.

Comment (r0mant) [inline tool/tctl/common/tctl.go:348]: Why is this change needed?

**Review by r0mant — CHANGES_REQUESTED** [review]
Congrats on your first PR @EdwardDowling!

In addition to the comments left on the code, please also include:

- Link to the ticket in the PR description.
- An example "before" and "after" behavior in the PR description.
- Test coverage.

File: tool/tctl/common/tctl.go (hunk 1 above)
Comment (EdwardDowling) [reply tool/tctl/common/tctl.go]: I've moved the setting of keysdir to the loadProfile function

File: tool/tsh/tctl_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			ccf:  &common.GlobalCLIFlags{},
+			cfg: &service.Config{
+				TeleportHome: "some/dir/that/does/not/exist",
+			},
+			want: trace.NotFound("profile is not found"),
+		}, {
+			name: "teleportHome is not specified",
+			ccf:  &common.GlobalCLIFlags{},
+			cfg:  &service.Config{},
+			want: trace.NotFound("profile is not found"),
```
Comment (r0mant) [inline tool/tsh/tctl_test.go:81]: Will this test case pass if you ran it on your local machine while you're logged in (i.e. when there's profile in `~/.tsh`)?

**Review by r0mant — COMMENTED** [review]
Looks good, just one question about the test.

File: tool/tsh/tctl_test.go (hunk 2 above)
Comment (EdwardDowling) [reply tool/tsh/tctl_test.go:81]: This test case will still pass and the correct error type is returned.

**Review by r0mant — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Comment by r0mant** [comment]
@EdwardDowling Can you backport this to `branch/v9`, `branch/v8` and then close the corresponding ticket?
